### PR TITLE
update to opentelemetry 0.24

### LINKED ISSUE #177: Support for opentelemetry 0.23.0
Hi 👋🏽

Would it be possible to add support for opentelemetry 0.23?

### PR DESCRIPTION
**Pull request by mellowagain**
fixes #177

# Checklist

- [x] Changelog updated

### CODE REVIEW

**Comment by mellowagain** [comment]
`tracing-opentelemetry` is only on otel 0.23, theres currently a pr open to update it to 0.24: https://github.com/tokio-rs/tracing-opentelemetry/pull/155

once that one is merged and released we can continue with the otel 0.24 upgrade. im pausing this pr for now

**Comment by mellowagain** [comment]
ran into another blocker which seems not that easily solveable, the `prometheus-exporter` feature depends on `opentelemetry_prometheus`, which is not updated to the new `opentelemetry` version and there doesn't even seem to be a plan to do so. on the readme of the project it states

> The development of prometheus exporter has halt until the Opentelemetry metrics API and SDK reaches 1.0. Current implementation is based on Opentelemetry API and SDK 0.23.

others people also running into this: https://github.com/open-telemetry/opentelemetry-rust/issues/1950

seems like this will be hard block for a while until `opentelemetry` reaches 1.0. im not sure if that is going to happen anytime soon, i thought it was blocked until there was consensus on https://github.com/open-telemetry/opentelemetry-rust/issues/1571 but that may be wrong.

we can optionally choose to fork and update `opentelemetry_prometheus` ourselves but that is additional maintainence burden and wouldn't play well if a user of `autometrics` includes the `opentelemetry_prometheus` themselves, as there would be conflicting versions then.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
-- 
+- `opentelemetry` has been updated to v0.24 (#179)
```
Comment (hatchan) [inline CHANGELOG.md:12]: Do we have to mark this as a breaking change?

File: examples/opentelemetry-push-custom/Cargo.toml [hunk 2]
```diff
@@ -5,12 +5,12 @@ publish = false
...
-autometrics = { path = "../../autometrics", features = ["opentelemetry-0_21"] }
+autometrics = { path = "../../autometrics", features = ["opentelemetry-0_24"] }
 autometrics-example-util = { path = "../util" }
 # Note that the version of the opentelemetry crate MUST match
 # the version used by autometrics
-opentelemetry = { version = "0.21", features = ["metrics"] }
-opentelemetry_sdk = { version = "0.21.1", features = ["metrics", "rt-tokio"] }
-opentelemetry-otlp = { version = "0.14", features = ["tonic", "metrics"] }
+opentelemetry = { version = "0.24", features = ["metrics"] }
+opentelemetry_sdk = { version = "0.24.1", features = ["metrics", "rt-tokio"] }
```
Comment (hatchan) [inline examples/opentelemetry-push-custom/Cargo.toml:13]: Do we have to use the `.1` here?

**Review by hatchan — APPROVED** [review]

**Pull request merged**